src/main/java/duke/command/DeleteCommand.java

```diff
@@ -0,0 +1,58 @@
+package duke.command;
+
+import duke.component.Parser;
+import duke.component.Storage;
+import duke.component.TaskList;
+import duke.component.Ui;
+import duke.task.Task;
+
+/**
+ * Represents a command for deleting a task.
+ */
+public class DeleteCommand extends Command {
+    /**
+     * Creates a command for deleting tasks.
+     * @param input the input command classified as DeleteCommand, starting with "delete "
+     */
+    public DeleteCommand(String input) {
+        super(input);
+    }
+
+    /**
+     * Executes the command, prints the result on ui and re-writes the source data file.
+     * @param ui the user interface object that is currently running
+     * @param list the current list of tasks
+     * @param storage the storage-writing object that is currently running
+     * @return the string that is to be printed on ui if the deleting is successful
+     * @throws InvalidCommandException if the input index for deleting is invalid
+     */
+    @Override
+    public String execute(Ui ui, TaskList list, Storage storage) throws InvalidCommandException {
+        int count = list.size();
+        int m = Parser.isValidDelete(input, count) - 1;
+        Task toDelete = list.get(m);
+        list.remove(toDelete);
+        storage.reWrite(list);
+        String temp = count <= 2 ? " task" : " tasks";
+        String str = "Noted. I've removed this task:\n\t    " + toDelete
```

Comment:
I like how you separated this long string concatenation into 2 lines, while leaving 8 spaces on the second line! This adheres to coding standards. Good job.